src/agent/src/main.c

```diff
@@ -1203,7 +1224,8 @@ int main(int argc, char** argv)
         {
             Log_Error("Agent health check failed.");
         }
-        return healthy ? 0 : 1;
+        ret = healthy ? 0 : 1;
```

Comment:
I think that if you initialize ret as 0, then you don't have to set it to zero everytime you goto done above. You only set it to 1 before the done tag